webapp/channels/src/packages/mattermost-redux/src/selectors/entities/users.ts

```diff
@@ -223,6 +223,27 @@ export const getCurrentUserMentionKeys: (state: GlobalState) => UserMentionKey[]
     },
 );
 
+export type UserHighlightKey = {
+    key: string;
+    caseSensitive?: boolean;
+}
+
+export const getUserHighlightKeys: (state: GlobalState) => UserHighlightKey[] = createSelector(
+    'getUserHighlightKeys',
+    getCurrentUser,
+    (user: UserProfile) => {
+        const highlightKeys: UserHighlightKey[] = [];
+
+        if (user.notify_props.highlight_keys.length > 0) {
+            user.notify_props.highlight_keys.split(',').forEach((key) => {
+                highlightKeys.push({key, caseSensitive: true});
```

Comment:
Speaking of that, I think we'd actually want these highlight keys to not be case sensitive

Reply:
+1 on keys not being case sensitive

cc: @abhijit-singh